sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileIO.java

```diff
@@ -979,6 +982,8 @@ public static FileNaming relativeFileNaming(
       abstract Builder<DestinationT, UserT> setSharding(
           PTransform<PCollection<UserT>, PCollectionView<Integer>> sharding);
 
+      abstract Builder<DestinationT, UserT> setMoveOption(StandardMoveOptions moveOption);
```

Comment:
What is the high level reason for making this customizable.
I think MoveOptions are internal implementation detail for FileIO that determines that it behaves correctly when workitems are retired by the runner.

Any changes to retry behavior could result in hard to track errors for the user when any failures are retried.